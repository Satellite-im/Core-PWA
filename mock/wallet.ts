export const Details = {
  balance: 328000,
  last_update: 1633500930632,
  currency: 'usd',
}

export const LinkedAccounts = [
  {
    type: 'credit-card',
    icon: 'credit-card',
    name: 'Discover Debit',
    number: '8312',
  },
  {
    type: 'debit-card',
    icon: 'credit-card',
    name: 'PNC Credit',
    number: '1234',
  },
  {
    type: 'hardware',
    icon: 'binary',
    name: 'Ledger Nano',
    number: '0000',
  },
]

export const RecentTransactions = [
  {
    id: '12345',
    user: {
      name: 'James Lightspeed',
      address: '0x07ee55aa48bb72dcc6e9d88856648910de513eca',
    },
    memo: 'Computer Game.',
    state: 'sent',
    at: 1633500930632,
    direction: 'out',
    amount: 3000,
  },
  {
    id: 'asdfewr21',
    user: {
      name: 'Derek Wormhole',
      address: '0x9bf4001de07dfd62b26a2f1307ee0c0307632d59',
    },
    memo: 'Thanks for lunch.',
    state: 'received',
    at: 1633500930632,
    amount: 82000,
  },
  {
    id: '1288roq',
    user: {
      name: 'Derek Wormhole',
      address: '0x9bf4001de07dfd62b26a2f1307ee0c0307632d59',
    },
    memo: '',
    state: 'requested',
    at: 1633500930632,
    amount: 2000,
  },
  {
    id: '12i9qwe',
    user: {
      name: 'Frank Meteor',
      address: '0xdc76cd25977e015ae17155770273ad58648913d3',
    },
    memo: '',
    state: 'sent',
    at: 1633500930632,
    amount: 9000,
  },
  {
    id: 'npqw012',
    user: {
      name: 'Frank Meteor',
      address: '0xdc76cd25977e015ae17155770273ad58648913d3',
    },
    memo: '',
    state: 'sent',
    at: 1633500930632,
    amount: 39000,
  },
]
