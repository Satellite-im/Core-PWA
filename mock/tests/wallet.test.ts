import { expect } from '@jest/globals'
import * as Wallet from '../wallet'

describe('init', () => {
  it('gets the Details constant', () => {
    expect(Wallet.Details).toMatchSnapshot()
  })
  it('gets the LinkedAccounts constant', () => {
    expect(Wallet.LinkedAccounts).toMatchSnapshot()
  })
  it('gets the RecentTransactions constant', () => {
    expect(Wallet.RecentTransactions).toMatchSnapshot()
  })
})
