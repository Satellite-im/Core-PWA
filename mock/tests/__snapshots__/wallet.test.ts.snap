// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init gets the Details constant 1`] = `
Object {
  "balance": 328000,
  "currency": "usd",
  "last_update": 1633500930632,
}
`;

exports[`init gets the LinkedAccounts constant 1`] = `
Array [
  Object {
    "icon": "credit-card",
    "name": "Discover Debit",
    "number": "8312",
    "type": "credit-card",
  },
  Object {
    "icon": "credit-card",
    "name": "PNC Credit",
    "number": "1234",
    "type": "debit-card",
  },
  Object {
    "icon": "binary",
    "name": "Ledger Nano",
    "number": "0000",
    "type": "hardware",
  },
]
`;

exports[`init gets the RecentTransactions constant 1`] = `
Array [
  Object {
    "amount": 3000,
    "at": 1633500930632,
    "direction": "out",
    "id": "12345",
    "memo": "Computer Game.",
    "state": "sent",
    "user": Object {
      "address": "0x07ee55aa48bb72dcc6e9d88856648910de513eca",
      "name": "James Lightspeed",
    },
  },
  Object {
    "amount": 82000,
    "at": 1633500930632,
    "id": "asdfewr21",
    "memo": "Thanks for lunch.",
    "state": "received",
    "user": Object {
      "address": "0x9bf4001de07dfd62b26a2f1307ee0c0307632d59",
      "name": "Derek Wormhole",
    },
  },
  Object {
    "amount": 2000,
    "at": 1633500930632,
    "id": "1288roq",
    "memo": "",
    "state": "requested",
    "user": Object {
      "address": "0x9bf4001de07dfd62b26a2f1307ee0c0307632d59",
      "name": "Derek Wormhole",
    },
  },
  Object {
    "amount": 9000,
    "at": 1633500930632,
    "id": "12i9qwe",
    "memo": "",
    "state": "sent",
    "user": Object {
      "address": "0xdc76cd25977e015ae17155770273ad58648913d3",
      "name": "Frank Meteor",
    },
  },
  Object {
    "amount": 39000,
    "at": 1633500930632,
    "id": "npqw012",
    "memo": "",
    "state": "sent",
    "user": Object {
      "address": "0xdc76cd25977e015ae17155770273ad58648913d3",
      "name": "Frank Meteor",
    },
  },
]
`;
